File: heart/src/main/java/com/vantagecircle/heartrate/core/HeartSupport.java
package com.vantagecircle.heartrate.core;

/**
 * Created by bapidas on 03/11/17.
 */

public interface HeartSupport {
    HeartSupport startPulseCheck(long timeLimit);
    HeartSupport startPulseCheck();
    HeartSupport addOnResultListener(PulseListener pulseListener);
    HeartSupport addOnFingerListener(FingerListener fingerListener);
    HeartSupport addOnTimerListener(TimerListener timerListener);
    void stopPulseCheck();
}

File: heart/src/main/java/com/bapidas/heartrate/core/HeartSupport.kt
package com.bapidas.heartrate.core

/**
 * Created by bapidas on 03/11/17.
 */
interface HeartSupport {
    fun startPulseCheck(timeLimit: Long): HeartSupport
    fun startPulseCheck(): HeartSupport
    fun addOnResultListener(pulseListener: PulseListener): HeartSupport
    fun addOnFingerListener(fingerListener: FingerListener): HeartSupport
    fun addOnTimerListener(timerListener: TimerListener): HeartSupport
    fun stopPulseCheck()
}